src/main/java/org/owasp/html/ExtensibleHtmlStreamRenderer.java

```diff
@@ -0,0 +1,510 @@
+            Encoding.encodeHtmlAttribOnto(value, output);
+            if (value.indexOf('`') != -1) {
+                // Apparently, in quirks mode, IE8 does a poor job producing innerHTML
+                // values. Given
+                // <div attr="``foo=bar">
+                // we encode &#96; but if JavaScript does:
+                // nodeA.innerHTML = nodeB.innerHTML;
+                // and nodeB contains the DIV above, then IE8 will produce
+                // <div attr=``foo=bar>
+                // as the value of nodeB.innerHTML and assign it to nodeA.
+                // IE8's HTML parser treats `` as a blank attribute value and foo=bar
+                // becomes a separate attribute.
+                // Adding a space at the end of the attribute prevents this by forcing
+                // IE8 to put double quotes around the attribute when computing
+                // nodeB.innerHTML.
+                output.append(' ');
+            }
+            output.append('"');
+        }
+
+        // Limit our output to the intersection of valid XML and valid HTML5 when
+        // the output contains no special HTML5 elements like <title>, <script>, or
+        // <textarea>.
+        if (HtmlTextEscapingMode.isVoidElement(elementName)) {
+            output.append(" /");
+        }
+
+        output.append('>');
+    }
+
+    public void closeTag(String elementName) {
+        try {
+            writeCloseTag(safeName(elementName));
+        } catch (IOException ex) {
+            ioExHandler.handle(ex);
+        }
+    }
+
+    protected void writeCloseTag(String uncanonElementName) throws IOException {
+        if (!open) {
+            throw new IllegalStateException();
+        }
+        String elementName = HtmlLexer.canonicalElementName(uncanonElementName);
+        if (!isValidHtmlName(elementName)) {
+            error("Invalid element name", elementName);
+            return;
+        }
+
+        if (pendingUnescaped != null) {
+            if (!lastTagOpened.equals(elementName)) {
+                error("Tag content cannot appear inside CDATA element", elementName);
+                return;
+            } else {
+                StringBuilder cdataContent = pendingUnescaped;
+                pendingUnescaped = null;
+                Encoding.stripBannedCodeunits(cdataContent);
+                int problemIndex = checkHtmlCdataCloseable(lastTagOpened, cdataContent);
+                if (problemIndex == -1) {
+                    if (cdataContent.length() != 0) {
+                        output.append(cdataContent);
+                    }
+                } /*
+                   * ZBUG-2547 : We have strictUnclosedCDATACheck retrieved from LC, if it is set
+                   * to false means we relax the check i.e. we don't throw an error rather based
+                   * on skipTagsUnclosedCdata from LC contains specific HTMl component
+                   * i.elementName we append to HTML
```

Comment:
Something is wrong with this description block. It does not read well.

Reply:
re modified the description , should be clear now